tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/PluginsSDKProjectProvider.java

```diff
@@ -112,78 +120,143 @@ public IStatus doCreateNewProject(
                 final String frameworkName = getFrameworkName( op );
 
                 workingDir = sdk.getLocation().append( ISDKConstants.PORTLET_PLUGIN_PROJECT_FOLDER ).toOSString();
-                baseDir = updateBaseDir ? workingDir : null;
 
-                newSDKProjectPath =
-                    sdk.createNewPortletProject(
-                        projectName, displayName, frameworkName, appServerProperties, separateJRE, workingDir, baseDir,
-                        monitor );
+                if( hasGradleTools )
+                {
+                    arguments.add( frameworkName );
+    
+                    sdk.createNewProject(
+                        projectName, arguments.toArray( new String[] {} ), "portlet", workingDir, monitor );
```

Comment:
Lets create the array just once at the top so we don't have to call "toArray()" so many times.